plugin.audio.talandaviad/addon.py

```diff
@@ -0,0 +1,92 @@
+import sys
+import urllib
+import urlparse
+import xbmcgui
+import xbmcplugin
+import requests
+from bs4 import BeautifulSoup
+import re
+
+
+def build_url(query):
+    base_url = sys.argv[0]
+    return base_url + '?' + urllib.urlencode(query)
+
+
+def get_page(url):
+    # download the source HTML for the page using requests
+    # and parse the page using BeautifulSoup
+    return BeautifulSoup(requests.get(url).text, 'html.parser')
+
+
+def parse_page(page):
+    songs = {}
+    index = 1
+
+    for item in page.find_all('item'):
+        songs.update({
+            index: {
+                'album_cover': re.search("src='([^']+)'", item.find('description').string).group(1),
+                'title': item.find('title').string,
+                'description': item.find('itunes:summary').string,
+                'url': item.find('guid').string
+            }
+        })
+        index += 1
+    return songs
+
+
+def build_song_list(songs):
+    song_list = []
+    for song in songs:
+        # create a list item using the song filename for the label
+        li = xbmcgui.ListItem(label=songs[song]['title'], thumbnailImage=songs[song]['album_cover'])
+        # set the fanart to the albumc cover
+        li.setProperty('fanart_image', '{0}/{1}'.format(ADDON_FOLDER, 'fanart.jpg'))
+        # set the list item to playable
+        li.setProperty('IsPlayable', 'true')
+        li.setProperty('ChannelName', 'ECO 99FM')
+        li.setProperty('PlotOutline', songs[song]['description'])
+        li.setInfo('video', {
+            'title': songs[song]['title'],
+            'genre': 'Podcast',
+            'plot': songs[song]['description']
+        })
+        li.setArt({
+            'thumb': songs[song]['album_cover'],
+            'poster': songs[song]['album_cover'],
+            'fanart': '{0}/fanart.jpg'.format(ADDON_FOLDER),
```

Comment:
please use `os.path.join` . Also note you'll have to change the location of the fanart within the addon folder structure as pointed out in another comment.